kotlinx-coroutines-core/common/src/EventLoop.common.kt

```diff
@@ -289,7 +289,11 @@ internal abstract class EventLoopImplBase: EventLoopImplPlatform(), Delay {
             // todo: we should unpark only when this delayed task became first in the queue
             unpark()
         } else {
-            DefaultExecutor.enqueue(task)
+            if (multithreadingSupported) {
```

Comment:
I guess we don't have enough time for a more structured approach, one that doesn't require this conditional, like ensuring that we can override `enqueue`?

Reply:
I'd rather hope for the deprecation of the old memory model and fast cleanup of every code path guarded by `multithreadingSupported`. That way it will be much simpler